index.bs

```diff
@@ -11302,6 +11306,136 @@ The [=remote end steps=] given |session| and |command parameters| are:
 </div>
 
 
+## The webExtensions Module ## {#module-webExtensions}
+
+The <dfn export for=modules>webExtensions</dfn> module contains functionality for
+managing and interacting with web-extensions.
+
+### Definition ### {#module-webExtensions-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+WebExtensionsCommand = (
+  webExtensions.Install
+)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+WebExtensionsResult = (
+  webExtensions.InstallResult
+)
+</pre>
+
+### Types ### {#module-webExtensions-types}
+
+#### The extensions.Extension Type #### {#type-webExtensions-Extension}
+
+<div algorithm>
+To <dfn>expand a extension data spec</dfn> given |extension data spec|:
+
+1. Let |installation type| be a value of <code>type</code> field of |extension data spec|.
+
+1. If installing |installation type| isn't supported return [=error=] with [=error code=] [=unsupported operation=].
+
+1. Switch on the value of the <code>installation type</code>:
+
+   <dl>
+    <dt>"<code>path</code>"
+    <dd>
+      1. Let |path| be a value of <code>path</code> field of |extension data spec|.
+      1. Let |entry| be a [=directory entry=] represeting the directory with that |path|.
+
+    <dt>"<code>archive-path</code>"
+    <dd>
+      1. Let |archive path| be a value of <code>path</code> field of |extension data spec|.
+      1. Perform any implementation-defined steps to extract ZIP from |archive path| and let |entry| by a [=directory entry=] representing the extraction result, if this fails return [=error=] with [=error code=] [=invalid web-extension=].
+
+    <dt>"<code>base64</code>"
+    <dd>
+      1. Let |archive path| be a value of <code>path</code> field of |extension data spec|.
+      1. Let |bytes| be [=forgiving-base64 decode=] |extension data spec|["<code>path</code>"].
+      1. Perform any implementation-defined steps to extract ZIP from |bytes| and let |entry| by a [=directory entry=] representing the extraction result, if this fails return [=error=] with [=error code=] [=invalid web-extension=].
+
+   </dl>
+
+1. Return [=success=] with data |entry|.
+
+</div>
+
+### Commands ### {#module-extensions-commands}
+
+#### The webExtensions.install Command #### {#command-extensions-install}
+
+The <dfn export for=commands>webExtensions.install</dfn> command installs an extension for the session.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      webExtensions.Install = (
+         method: "webExtensions.install",
+         params: webExtensions.InstallParameters
+      )
+
+      webExtensions.InstallParameters = {
+         extensionData: webExtensions.ExtensionPath / webExtensions.ExtensionArchivePath / webExtensions.ExtensionBase64Encoded,
+         ? allowPrivateBrowsing: bool .default false,
+      }
```

Comment:
I left some thoughts on this above. I don't think we should add a `temporary` property that controls the loading of unpacked extensions since that couples two unrelated concepts.

Reply:
Sounds fine when not all browsers support this type of installation. We will then add such a field for Firefox only.